src/Xamarin.Android.Build.Tasks/Utilities/TypeMapGenerator.cs

```diff
@@ -502,7 +569,7 @@ void OutputModule (BinaryWriter bw, ModuleDebugData moduleData)
 			foreach (TypeMapDebugEntry entry in moduleData.JavaToManagedMap) {
 				bw.Write (outputEncoding.GetBytes (entry.JavaName));
 				PadField (bw, entry.JavaName.Length, (int)moduleData.JavaNameWidth);
-				bw.Write (entry.ManagedIndex);
+				bw.Write (entry.SkipInJavaToManaged ? UInt32.MaxValue : (uint)entry.ManagedIndex);
```

Comment:
This makes it look like we're using `uint.MaxValue` as a "placeholder" value for an "invalid" mapping.  If so, we should semantically name this constant:

```csharp
const uint InvalidJavaToManagedMappingIndex = uint.MaxValue;
```